src/ReactiveUI.XamForms.Tests/RoutedViewHostTest.cs

```diff
@@ -0,0 +1,304 @@
+            Locator.CurrentMutable.Register<IViewFor<ChildViewModel>>(() => new ChildView());
+
+            Locator.CurrentMutable.Register<IScreen>(() => _navigationViewModel);
+            Locator.CurrentMutable.Register<IRoutableViewModel>(() => new MainViewModel(), nameof(MainViewModel));
+            Locator.CurrentMutable.Register<IRoutableViewModel>(() => new ChildViewModel(), nameof(ChildViewModel));
+        }
+
+        [Fact]
+        public void NewRoutedViewHostHasNoCurrentPage()
+        {
+            var fixture = CreateRoutedViewHost(initialViewModel: null);
+
+            Assert.Null(fixture.CurrentPage);
+            Assert.Equal(0, fixture.StackDepth);
+        }
+
+        [Fact]
+        public void NewRoutedViewHostHasMainViewCurrentPage()
+        {
+            var fixture = CreateRoutedViewHost();
+
+            var currentPage = fixture.CurrentPage;
+            Assert.NotNull(currentPage);
+            Assert.IsType<MainView>(currentPage);
+            Assert.IsType<MainViewModel>(currentPage.BindingContext);
+            Assert.NotNull(currentPage.BindingContext);
+            Assert.Equal("Main view", currentPage.Title);
+
+            Assert.Equal(1, fixture.StackDepth);
+        }
+
+        [Fact]
+        public async Task NavigateToMainViewFromNoPage()
+        {
+            var fixture = CreateRoutedViewHost(initialViewModel: null);
+
+            var viewModel = await _navigationViewModel.Navigate(nameof(MainViewModel)).GetAwaiter();
+
+            Assert.NotNull(viewModel);
+
+            var currentPage = fixture.CurrentPage;
+            Assert.NotNull(currentPage);
+            Assert.IsType<MainView>(currentPage);
+            Assert.IsType<MainViewModel>(currentPage.BindingContext);
+            Assert.Equal(viewModel, currentPage.BindingContext);
+            Assert.Equal("Main view", currentPage.Title);
+
+            Assert.Equal(1, fixture.StackDepth);
+        }
+
+        [Fact]
+        public async Task NavigateToChildView()
+        {
+            var fixture = CreateRoutedViewHost();
+
+            var viewModel = await _navigationViewModel.Navigate(nameof(ChildViewModel)).GetAwaiter();
+
+            Assert.NotNull(viewModel);
+
+            var currentPage = fixture.CurrentPage;
+            Assert.NotNull(currentPage);
+            Assert.IsType<ChildView>(currentPage);
+            Assert.IsType<ChildViewModel>(currentPage.BindingContext);
+            Assert.Equal(viewModel, currentPage.BindingContext);
+            Assert.Equal("Child view: ", currentPage.Title);
+
+            Assert.Equal(2, fixture.StackDepth);
+        }
+
+        [Fact]
+        public async Task NavigateToSecondChildView()
+        {
+            var fixture = CreateRoutedViewHost();
+
+            await _navigationViewModel.Navigate(nameof(ChildViewModel)).GetAwaiter();
+            var viewModel = await _navigationViewModel.NavigateToChild("Testing").GetAwaiter();
```

Comment:
Probably worth removing the calls to `GetAwaiter()`. Usually, you can just `await` an `IObservable`.

Reply:
Good point. Thank you. Removed.